### ISSUE TITLE
Make requires python to be compiled with openssl

### ISSUE THREAD

**Issue by MajdiRadaideh** [mit-quest/nuclear-core-design#12]
[Note sure of what causing the attached error, I just ran make command without any options as given in the instructions. My python version is 3.6.7, my pip version is 8.1.1. Thanks!
![make_error](https://user-images.githubusercontent.com/37671304/70550815-fb381b00-1b3b-11ea-84e3-07d03d4117e9.png)

**Comment by IsaacWolverton** [comment]
This seems to be due to the fact that anaconda has an outdated version of ssl: [anaconda-issues](https://github.com/ContinuumIO/anaconda-issues/issues/10592). Can you try running `sudo apt-get upgrade openssl`  and then running make agian?

**Comment by MajdiRadaideh** [comment]
Error stays. I tried to compile with the raw python2 in the system, no luck. Honestly, I did not dive into this issue thinking it is something easy to fix. I will do that as I guess it may have to do with my pc. If you got other ideas, plz let me know.

**Comment by Josh-Joseph** [comment]
Bit more info may be helpful @MajdiRadaideh:
- did you do a `make clean` before retrying make?
- what version of ubuntu are you running?
- when you ran the upgrade openssl Isaac suggested did it actually update openssl or it just said you were already at the most recent version?

**Comment by MajdiRadaideh** [comment]
1- I do clean every time before make, nothing happens. I actually reclone the package again, still no luck. 
2- 16.04
3- It did updated openssl, I have not updated anaconda for a while so it spent sometime indeed to get this done.

**Comment by MajdiRadaideh** [comment]
I got the first issue resolved by upgrading virtual env, i got the flow to move a little bit
`pip install --upgrade virtualenv`

but now openssl error is coming in, which I think what Issac was talking about 
![image](https://user-images.githubusercontent.com/37671304/70567870-5c6fe680-1b5c-11ea-9c42-92a1b8e8c318.png)

**Comment by Josh-Joseph** [comment]
Yeah it looks like your python was not compiled withe libraries you needed -- see if these are any help:

https://askubuntu.com/questions/1164352/cannot-use-pip-ubuntu-pip-is-configured-with-locations-that-require-tls-ssl

https://stackoverflow.com/questions/45954528/pip-is-configured-with-locations-that-require-tls-ssl-however-the-ssl-module-in

**Comment by MajdiRadaideh** [comment]
My updates on this issue after series of trials, it turns out that @Josh-Joseph direction was the best. Here is everything I did to solve the problem.

```
pip install --upgrade virtualenv
sudo apt install libssl-dev libncurses5-dev libsqlite3-dev libreadline-dev libtk8.5 libgdm-dev libdb4o-cil-dev libpcap-dev
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt-get update
sudo apt-get install python3.6
sudo apt-get install python3.6-dev

#Edit the MakeFile to use the new python3.6, not Anaconda python or system python (if they exist)
PYTHON ?= /usr/bin/python3.6

make
```

**Comment by MajdiRadaideh** [comment]
A minor typo was observed in the file extension `config.yaml`, which is given as `config.yml` in the `random_agent.py`. Other than that all API tests are working nicely now, thanks guys for the help :)

**Comment by Josh-Joseph** [comment]
nice. glad to hear it -- yeah apologies about that renaming bug. It's fixed in #11 so once that gets merged it should be all good
